uni-bern-institut-fur-musikwissenschaft-note.csl

```diff
@@ -0,0 +1,543 @@
+    </group>
+  </macro>
+  <macro name="publisher_location_year">
+    <group delimiter=", ">
+      <text macro="original-publisher_location_year"/>
+      <text macro="current-publisher_location_year"/>
+    </group>
+  </macro>
+  <macro name="original-publisher_location_year">
+    <choose>
+      <if match="any" variable="original-publisher original-publisher-place">
+        <choose>
+          <if type="article-journal article-magazine article-newspaper motion_picture" match="none">
+            <group delimiter=" ">
+              <group delimiter=": ">
+                <text macro="original-publisher-place"/>
+                <text variable="original-publisher"/>
+              </group>
+              <text macro="origyear-reprint"/>
+            </group>
+          </if>
+        </choose>
+      </if>
+    </choose>
+  </macro>
+  <macro name="current-publisher_location_year">
+    <choose>
+      <if type="article-journal article-magazine article-newspaper motion_picture" match="none">
+        <group delimiter=" ">
+          <text macro="reprint-term"/>
+          <group delimiter=": ">
+            <text macro="publisher-place"/>
+            <text variable="publisher"/>
+          </group>
+          <text macro="year"/>
+        </group>
+      </if>
+    </choose>
+  </macro>
+  <macro name="publisher-place">
+    <choose>
+      <if match="none" variable="publisher-place">
+        <choose>
+          <if type="book chapter entry-encyclopedia manuscript pamphlet" match="any">
+            <text value="o.O."/>
+          </if>
+        </choose>
+      </if>
+      <else>
+        <choose>
+          <if type="motion_picture" match="none">
+            <text variable="publisher-place"/>
+          </if>
+        </choose>
+      </else>
+    </choose>
+  </macro>
+  <macro name="original-publisher-place">
+    <choose>
+      <if match="none" variable="original-publisher-place">
+        <choose>
+          <if type="book chapter entry-encyclopedia manuscript pamphlet" match="any">
+            <text value="o.O."/>
+          </if>
+        </choose>
+      </if>
+      <else>
+        <choose>
+          <if type="motion_picture" match="none">
+            <text variable="original-publisher-place"/>
+          </if>
+        </choose>
+      </else>
+    </choose>
+  </macro>
+  <macro name="year">
+    <choose>
+      <if match="none" variable="issued">
+        <choose>
+          <if type="book chapter entry-encyclopedia manuscript thesis pamphlet" match="any">
+            <text macro="no-date"/>
+          </if>
+        </choose>
+      </if>
+      <else-if type="interview" match="any">
+        <date form="numeric" variable="issued"/>
+      </else-if>
+      <else>
+        <group>
+          <number vertical-align="sup" variable="edition"/>
+          <date date-parts="year" form="text" variable="issued"/>
+        </group>
+      </else>
+    </choose>
+  </macro>
+  <macro name="no-date">
```

Comment:
I don't really see the use of single-line macros. They likely have an (albeit tiny) negative effect on performance, increase style length and decrease legibility.

Reply:
Ok. That's gone as well.